packages/@sanity/cli/src/actions/init-project/initProject.ts

```diff
@@ -202,9 +226,205 @@ export default async function initSanity(
   // Ensure we are using the output path provided by user
   outputPath = answers.outputPath
 
+  if (initNext) {
+    const useTypeScript = await prompt.single<boolean>({
+      type: 'confirm',
+      message: `Would you like to use TypeScript?`,
+      default: true,
+    })
+
+    const fileEnding = useTypeScript ? 'ts' : 'js'
+
+    const embeddedStudio = await prompt.single({
+      type: 'confirm',
+      message: `Would you like an embedded Sanity Studio?`,
+      default: true,
+    })
+
+    if (embeddedStudio) {
+      const useAppDir = await prompt.single({
+        type: 'confirm',
+        message: `Would you like to use the Next.js app dir?`,
+        default: false,
+      })
+
+      // find source path (app or pages dir)
+      const srcDir = useAppDir ? 'app' : 'pages'
+      let srcPath = path.join(workDir, srcDir)
+
+      if (!existsSync(srcPath)) {
+        srcPath = path.join(workDir, 'src', srcDir)
+        if (!existsSync(srcPath)) {
+          await fs
+            .mkdir(srcPath, {recursive: true})
+            .catch(() => debug('Error creating folder %s', srcPath))
+        }
+      }
+
+      const studioPath = await prompt.single({
+        type: 'input',
+        message: 'What url do you want to use for the embedded studio?',
+        default: '/studio',
+        validate(input) {
+          if (!input.startsWith('/')) {
+            return 'Must start with /'
+          }
+
+          if (input.endsWith('/')) {
+            return 'Must not end with /'
+          }
+
+          // a-Z, 0-9, -, _ and /
+          if (!/^[a-zA-Z0-9-_\\/]+$/.test(input)) {
+            return 'Must only contain a-Z, 0-9, -, _ and /'
+          }
+
+          return true
+        },
+      })
+
+      const embeddedStudioPath = path.join(
+        srcPath,
+        `${studioPath}/`,
+        useAppDir ? '[[..index]]/page.tsx' : '[[...index]].tsx'
+      )
+
+      await writeOrOverwrite(
+        embeddedStudioPath,
+        (useAppDir ? sanityStudioAppTemplate : sanityStudioPagesTemplate).replace(
+          ':configPath:',
+          new Array(embeddedStudioPath.slice(workDir.length).split('/').length - 1)
+            .join('../')
+            .concat('sanity.config')
+        )
+      )
+
+      const sanityConfigPath = path.join(workDir, 'sanity.config.'.concat(fileEnding))
+      await writeOrOverwrite(
+        sanityConfigPath,
+        sanityConfigTemplate
+          .replace(':route:', embeddedStudioPath.slice(workDir.length).replace('src/', ''))
+          .replace(':basePath:', studioPath)
+      )
+    }
+
+    const sanityCliPath = path.join(workDir, 'sanity.cli.'.concat(fileEnding))
+    await writeOrOverwrite(sanityCliPath, sanityCliTemplate)
+
+    // write sanity folder files
+    const recursivelyWriteFiles = async (
```

Comment:
This is used to recursively write the object tree of files depending on which template you use.